### PR TITLE
JAMES-3015 Deployment tests with maven

### PR DESCRIPTION
**Pull request by mbaechler**
This PR purpose is to check the behavior of the current state of this work on jenkins.

It is not intended to be validated and/or merged as there's some more work to do before reaching merge point.

### CODE REVIEW

File: server/container/guice/cassandra-guice/src/main/jib/glowroot/admin.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "web": {
+    "bindAddress": "0.0.0.0"
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/src/main/jib/glowroot/admin.json]: Can't we share all glowroot conf in a single maven module?

File: server/app/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Run James
```
Comment (chibenwa) [inline server/app/Dockerfile]: Shouldn't we remove the one in /dockerfiles/ and update the README?

**Comment by chibenwa** [comment]
Why not run deployment test this way for other products? (I bet it is next steps ^^)

**Comment by chibenwa** [comment]
If I run tests with `-Djib.skip` deployment tests should be skept. Is it the case today or are we failing?

File: server/container/guice/cassandra-guice/src/main/jib/glowroot/admin.json (hunk 1 above)
Comment (rouazana) [reply server/container/guice/cassandra-guice/src/main/jib/glowroot/admin.json]: no the glowroot conf is not exactly the same (think of blob store not suitable for memory)

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java [hunk 3]
```diff
@@ -58,14 +62,23 @@ public InitializationStatus initializeTables() {
...
-        Flux.fromIterable(module.moduleTables())
+        Flux.fromIterable(listAllTables(executor))
                 .publishOn(Schedulers.elastic())
-                .map(CassandraTable::getName)
                 .flatMap(name -> truncate(executor, name))
                 .then()
                 .block();
     }
 
+    private List<String> listAllTables(CassandraAsyncExecutor executor) {
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java]: We should not change the default behavior of the Cassandra extension.
We can, for now, either generate a new keyspace for each smoketest or recreate the keyspace between two runs.

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraClusterExtension.java [hunk 4]
```diff
@@ -49,7 +58,8 @@ public void beforeAll(ExtensionContext extensionContext) {
     }
 
     private void start() {
-        cassandraCluster = CassandraCluster.create(cassandraModule, cassandraExtension.getDockerCassandra().getHost());
+        cassandraCluster = keyspace.map(kspace -> CassandraCluster.create(cassandraModule, cassandraExtension.getDockerCassandra().getHost(), kspace))
```
Comment (mbaechler) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraClusterExtension.java]: what about choosing the default keyspace name here?

File: mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/DockerDeploymentValidationGuiceCassandraIT.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+public class DockerDeploymentValidationGuiceCassandraIT implements DeploymentValidation {
+    private static final Network network = Network.newNetwork();
+    private static final DockerCassandra cassandraContainer = new DockerCassandra("cassandra_3_11_3",
+        DockerCassandra.AdditionalDockerFileStep.IDENTITY,
+        container -> container.withNetworkAliases("cassandra")
+            .withNetwork(network));
+    private static final DockerCassandraExtension cassandraExtension = new DockerCassandraExtension(cassandraContainer);
+    public static final String KEYSPACE = "apache_james";
+    @RegisterExtension
+    @Order(1)
```
Comment (mbaechler) [inline mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/DockerDeploymentValidationGuiceCassandraIT.java]: why do you need that?

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java [hunk 6]
```diff
@@ -19,16 +19,20 @@
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java]: this changeset must be split in several pieces, there are very different things here, no so hard to rework

**Comment by chibenwa** [comment]
Many merge conflict, inactive for very long, opened on linagora repo and not the ASF one...

Reopen on ASF repository if relevant.

**Pull request closed without merging**